### PR TITLE
[ Example ] Tizen C API Train with Generator Example

### PR DESCRIPTION
**Pull request by jijoongmoon**
Demo example to train with Tizen C API Train with generator.

Add simple example to train 1 FC Layer neural network and get the
train & validation data from generator created by user.

Resolves:

**Self evaluation:**
1. Build test:	 [X]Passed [ ]Failed [ ]Skipped
2. Run test:	 [X]Passed [ ]Failed [ ]Skipped

Signed-off-by: jijoong.moon <jijoong.moon@samsung.com>

### CODE REVIEW

File: Applications/Tizen_CAPI/capi_func.c [hunk 1]
```diff
@@ -0,0 +1,340 @@
...
+
+  if (id < 0)
+    return false;
+
+  position = (feature_size + num_class) * id * sizeof (float);
+  if (position > file_length || position > ULLONG_MAX) {
+    return false;
+  }
+
+  F = fopen (file_name, "rb");
```
Comment (dongju-chae) [inline Applications/Tizen_CAPI/capi_func.c:94]: How about checking the validity of `F`?

File: Applications/Tizen_CAPI/capi_func.c [hunk 2]
```diff
@@ -0,0 +1,340 @@
...
+    alloc_train = true;
+  }
+
+  for (i = 0; i < data_size; i++) {
+    if (!duplicate[i])
+      count++;
+  }
+
+  if (count < mini_batch) {
+    free (duplicate);
```
Comment (dongju-chae) [inline Applications/Tizen_CAPI/capi_func.c:159]: Need to check `nullptr` here?

File: Applications/Tizen_CAPI/capi_func.c (hunk 1 above)
Comment (jijoongmoon) [reply Applications/Tizen_CAPI/capi_func.c:95]: Fixed. Thanks.

File: Applications/Tizen_CAPI/capi_func.c (hunk 2 above)
Comment (jijoongmoon) [reply Applications/Tizen_CAPI/capi_func.c:172]: Fixed. Thanks.

**Pull request merged**